.github/workflows/release.yml

```diff
@@ -91,7 +94,7 @@ jobs:
 
       - name: Create GitHub release
         env:
-          GITHUB_TOKEN: ${{ secrets.BOT_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
@jack-berg I think this change fixes https://github.com/open-telemetry/opentelemetry-java/runs/6356816939?check_suite_focus=true

```
HTTP 404: Not Found (https://api.github.com/repos/open-telemetry/opentelemetry-java/releases)
```